integrated_channels/integrated_channel/exporters/learner_data.py

```diff
@@ -608,39 +674,24 @@ def _collect_grades_data(self, enterprise_enrollment, course_details, is_audit_e
             course_details (dict): the course details for the course in the enterprise enrollment record.
 
         Returns:
-            completed_date: Date the course was completed, this is None if course has not been completed.
+            completed_date: Date the course was completed, None if course has not been completed.
             grade: Current grade in the course.
             is_passing: Boolean indicating if the grade is a passing grade or not.
+            percent_grade: a number between 0 and 100
         """
-        if self.grades_api is None:
-            self.grades_api = GradesApiClient(self.user)
 
         course_id = enterprise_enrollment.course_id
         username = enterprise_enrollment.enterprise_customer_user.user.username
+        user_id = enterprise_enrollment.enterprise_customer_user.user_id
 
-        try:
-            grades_data = self.grades_api.get_course_grade(course_id, username)
-
-        except HttpNotFoundError as error:
-            # Grade not found, so we have nothing to report.
-            if hasattr(error, 'response'):
-                response_content = error.response.json()  # pylint: disable=no-member
-                if response_content.get('error_code', '') == 'user_not_enrolled':
-                    # This means the user has an enterprise enrollment record but is not enrolled in the course yet
-                    LOGGER.info(
-                        '[Integrated Channel] User is not enrolled in the course.'
-                        ' Course: {course_id}, EnterpriseEnrollment: {enterprise_enrollment},'
-                        ' Username: {username}'.format(
-                            course_id=course_id,
-                            username=username,
-                            enterprise_enrollment=enterprise_enrollment.pk))
-                    return None, None, None, None
+        grades_data = get_single_user_grade(course_id, username)
```

Comment:
Again, can we  use user_id to get this information instead of username?

Reply:
ah not sure thatt will work given the api call I am using is expecting a username it appears

```
CourseGradeFactory().read(grade_user, course_key=course_key)
```

Reply:
That's a "user" as in the object "Auth.user", not a string. I think the user id should be used to get you a user and username shouldn't work anyway, let me look into that for a second.

Reply:
Alright so the original call you replaced used the username for who knows what reason. But you don't have to do that, and anyway the grade_user you are looking at there is definitely a user object, not a username string. So you can use django to get the correct user by id and pass it in, and just remove the username usage here completely.

Reply:
got it, now passing the user object instead!